warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/shard/FieldDateShardIdGenerator.java

```diff
@@ -0,0 +1,417 @@
+package datawave.ingest.mapreduce.handler.shard;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.collect.Multimap;
+
+import datawave.ingest.data.RawRecordContainer;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.util.time.DateHelper;
+
+/**
+ * FieldDateShardIdGenerator ia an implementation of {@link ShardIdGenerator} that will shift the range of shard IDs based on an event date range and fields
+ * within the record. For events that do not match one of the configured fields, they will be remapped to a base set of shards. It is expected that the
+ * configured shards for that day will be set to the total number of shards possible for query purposes.
+ *
+ * So for example:
+ *
+ * @formatter:off
+ * shardIdFactory.generator.1 = datawave.ingest.mapreduce.handler.shard.FieldDateShardIdGenerator
+ * shardIdFactory.generator.1.baseNumShards = 20
+ * shardIdFactory.generator.1.startDate = 20250101000000
+ * shardIdFactory.generator.1.endDate = 20250103235959
+ * shardIdFactory.generator.1.datatype.1 = DT1
+ * shardIdFactory.generator.1.field.1 = FIELD1
+ * shardIdFactory.generator.1.regex.1 = REGEX1
+ * shardIdFactory.generator.1.shards.1 = 10
+ * shardIdFactory.generator.1.datatype.2 = DT2
+ * shardIdFactory.generator.1.field.2 = FIELD2
+ * shardIdFactory.generator.1.regex.2 = REGEX2
+ * shardIdFactory.generator.1.shards.2 = 15
+ * @formatter:on
```

Comment:
```suggestion
 * <pre>
 * {@code
 * shardIdFactory.generator.1 = datawave.ingest.mapreduce.handler.shard.FieldDateShardIdGenerator
 * shardIdFactory.generator.1.baseNumShards = 20
 * shardIdFactory.generator.1.startDate = 20250101000000
 * shardIdFactory.generator.1.endDate = 20250103235959
 * shardIdFactory.generator.1.datatype.1 = DT1
 * shardIdFactory.generator.1.field.1 = FIELD1
 * shardIdFactory.generator.1.regex.1 = REGEX1
 * shardIdFactory.generator.1.shards.1 = 10
 * shardIdFactory.generator.1.datatype.2 = DT2
 * shardIdFactory.generator.1.field.2 = FIELD2
 * shardIdFactory.generator.1.regex.2 = REGEX2
 * shardIdFactory.generator.1.shards.2 = 15
 * }
 * </pre>
```
This should resolve your build failure that relates to the `@formatter` tags. This presents the following lines as a code block.